core/corehttp/gateway_handler.go

```diff
@@ -92,16 +92,24 @@ func (i *gatewayHandler) getOrHeadHandler(w http.ResponseWriter, r *http.Request
 
 	urlPath := r.URL.Path
 
+	// If the gateway is behind a reverse proxy and mounted at a sub-path,
+	// the prefix header can be set to signal this sub-path.
+	// It will be prepended to links in directory listings and the index.html redirect.
+	prefix := ""
+	if prefixHdr := r.Header["X-Ipfs-Gateway-Prefix"]; len(prefixHdr) > 0 {
+		log.Debugf("X-Ipfs-Gateway-Prefix: %s", prefixHdr[0])
+		prefix = prefixHdr[0]
+	}
```

Comment:
this SGTM. though not sure if this gives anyone power by putting it as a req header. cant think of an attack now, but there may be one.